File: binary_search/problems/977_squares_of_a_sorted_array.py
# Given an integer array nums sorted in non-decreasing order, return an array of the squares of each number sorted in non-decreasing order.

class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        
        for i in range(len(nums)):
            nums[i] = nums[i] ** 2
        
        nums.sort()
        return nums

class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        
        l, r = 0, len(nums) - 1
        ans = deque()
        while l <= r:
            left = nums[l] ** 2
            right = nums[r] ** 2
            if left > right:
                ans.appendleft(left)
                l += 1
            else:
                ans.appendleft(right)
                r -= 1
        return ans

File: typescript/problems/977_squares_of_a_sorted_array.ts
/**
 Given an integer array nums sorted in non-decreasing order, return an array of the squares of each number sorted in non-decreasing order.
 */

function sortedSquares(nums: number[]): number[] {
    let ans: number[] = [];

    for (const n of nums) {
        ans.push(n**2)
    }
    ans.sort((a, b) => a - b);
    return ans
};